entrypoint.sh

```diff
@@ -80,11 +80,18 @@ pip3 install -r /tmp/gzdev/requirements.txt --break-system-packages
 apt-get update 2>&1
 echo ::endgroup::
 
-echo ::group::Install tools: pip
-pip3 install -U vcstool colcon-common-extensions --break-system-packages
-echo ::endgroup::
-
 if [ -f "$SOURCE_DEPENDENCIES" ] || [ -f "$SOURCE_DEPENDENCIES_VERSIONED" ] ; then
+  echo ::group::Prepare colcon and vcs for source dependencies
+  # Add colcon repository
+  mkdir -p /etc/apt/keyrings/
+  for tool in colcon vcstool; do
+    curl -fsSL "https://packagecloud.io/dirk-thomas/${tool}/gpgkey" | gpg --dearmor > "/etc/apt/keyrings/${tool}-archive-keyring.gpg"
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/${tool}-archive-keyring.gpg] https://packagecloud.io/dirk-thomas/${tool}/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/${tool}.list > /dev/null
```

Comment:
I see that `vcstool` is coming from a third-party repo as well. This probably what we want since the one in Ubuntu upstream is usually outdated, but the statement in the PR description mentions that we're using the one from upstream, so I was a little confused.